### PR TITLE
[MERGE] event(_*): improve attendees views and reporting

### PR DESCRIPTION
**Pull request by ija-odoo**
PURPOSE

To improve Event attendees views and reporting for better usability.

SPECIFICATIONS

The goal of this commit is to improve attendees views and reporting in
Event, the changes  done are listed below.

- When opening attendees from an event (stat button or kanban button), it will display 
   the list view (before the kanban one) and group by create_date (week).
- Removed the date_open field from the model as it is a duplicate of the create_date.
- On the attendees tree view, displayed the registration_answer_ids field with a 
   many2many_tags widget ​(optional default hide), before the Status.
- Displayed the "X Cancel" button of the attendees tree view in red.
- Moved the partner_id field next to the ticket and kept it editable in attendees form view,
   also moved visitor field below the Mobile one so that even in debug mode, Attendee
   Name is the "primary" field.
- Added a badge widget on the `state` field.
- Added next activity widget to the field `event_ticket_id`.

While coming to Attendee reporting section we have changed the default view as graph
view instead of stern pivot view and added default groupby `state`, `event_id`, `create_date` and 
created a new filter `Last 30 Days` and made it default.
Apart from that we have made `summary` field default hide in Meeting Room tree view.

LINKS
PR #79166
Task 2670765

### CODE REVIEW

File: addons/event/views/event_views.xml [hunk 1]
```diff
@@ -495,7 +495,7 @@
             <field name="name">event.registration.tree</field>
             <field name="model">event.registration</field>
             <field name="arch" type="xml">
-                <tree string="Registration" multi_edit="1" sample="1">
+                <tree string="Registration" multi_edit="1" sample="1" class="o_event_attendee_view_tree">
```
Comment (tde-banana-odoo) [inline addons/event/views/event_views.xml:498]: Would name it o_event_registration_view_tree, to be model-based (as attendee is the "business name").

File: addons/website_event_questions/models/event_registration.py [hunk 2]
```diff
@@ -28,3 +28,10 @@ class EventRegistrationAnswer(models.Model):
     _sql_constraints = [
         ('value_check', "CHECK(value_answer_id IS NOT NULL OR COALESCE(value_text_box, '') <> '')", "There must be a suggested value or a text value.")
     ]
+
+    # for displaying selected answers by attendees in attendees list view
+    def name_get(self):
+        res = []
+        for rec in self:
+            res.append((rec.id, ("%s" % rec.value_answer_id.name)))
```
Comment (tde-banana-odoo) [inline addons/website_event_questions/models/event_registration.py:36]: Well, shouldn't we either take value_answer_id or value_text_box ? I don't remember all details of this model, but if I remember well either this is predefined, either user entry ? In that case, how is it displayed with your name_get ?

File: addons/website_event_questions/views/event_registration_views.xml [hunk 3]
```diff
@@ -24,4 +24,15 @@
             </sheet>
         </field>
     </record>
+
+    <record id="view_event_registration_tree" model="ir.ui.view">
```
Comment (tde-banana-odoo) [inline addons/website_event_questions/views/event_registration_views.xml:28]: New views should be correctly xml-id-ed, so "event_registration_view_tree" :) (even if base view is still using an old formatting)

File: addons/event/views/event_views.xml [hunk 4]
```diff
@@ -533,7 +533,6 @@
                         <div class="oe_button_box" name="button_box"/>
                         <group>
                             <group string="Attendee" name="attendee">
-                                <field name="partner_id" attrs="{'readonly':[('state', '!=', 'draft')]}"/>
```
Comment (tde-banana-odoo) [inline addons/event/views/event_views.xml:536]: Actually field was duplicated ... could be great to explain it in commit message (I thought you were removing it ^^ ) . By the way, don't we keep the attrs ?

File: addons/event/views/event_views.xml [hunk 5]
```diff
@@ -698,6 +698,7 @@
                         domain="[('my_activity_date_deadline', '=', context_today().strftime('%Y-%m-%d'))]"/>
                     <filter invisible="1" string="Future Activities" name="activities_upcoming_all"
                         domain="[('my_activity_date_deadline', '&gt;', context_today().strftime('%Y-%m-%d'))]"/>
+                    <filter string="Last 30 days" name="filter_last_month_activities" domain="[('create_date','&gt;', (context_today() - datetime.timedelta(days=30)).strftime('%Y-%m-%d'))]"/>
```
Comment (tde-banana-odoo) [inline addons/event/views/event_views.xml:701]: filter_last_month_creation maybe ? As it is not linked to write_date, just the creation date ...

File: addons/event_sms/tests/test_sms_schedule.py [hunk 6]
```diff
@@ -45,8 +45,10 @@ def setUpClass(cls):
         })
 
     def test_sms_schedule(self):
+        # forcefully mimic the real life scenario where registrations are done few days
+        # before the event begins (for checking subscription scheduler)
         with self.mockSMSGateway():
-            self._create_registrations(self.event_0, 3)
+            self._create_registrations(self.event_0, 3, self.event_0.date_begin + relativedelta(days=-5))
```
Comment (tde-banana-odoo) [inline addons/event_sms/tests/test_sms_schedule.py:51]: Why this change actually ? Not sure to understand

**Review by tde-banana-odoo — COMMENTED** [review]
Validation ! Seems globally good, some small comments :)

File: addons/event/views/event_views.xml (hunk 4 above)
Comment (ija-odoo) [reply addons/event/views/event_views.xml:536]: It is already mentioned in the commit that the field has been changed its position next to the ticket and we don't need to keep the attrs because in the specification it clearly mentioned to keep this field editable.

File: addons/website_event_questions/models/event_registration.py (hunk 2 above)
Comment (ija-odoo) [reply addons/website_event_questions/models/event_registration.py:36]: we only need to take value_answer_id, in the specification it mentioned that no need to display the text input in the list view.

File: addons/event/tests/test_event_flow.py [hunk 7]
```diff
@@ -14,30 +14,32 @@
 class TestEventUI(TestEventCommon):
 
     def test_event_registration_partner_sync(self):
-        """ Ensure onchange on partner_id is kept for interface, not for computed
```
Comment (tde-banana-odoo) [inline addons/event/tests/test_event_flow.py:17]: Actually ``test_registration_partner_sync`` already checks most of the compute, if not everything. If we remove the onchange, I feel like this whole test could be removed as it is linked only with the ochange. You should probably tell it is a revert of odoo/odoo@c6003af73409e2cd124f641fb7714fefa9dc8b01 since we feel this is not necessary anymore.

File: addons/event/views/event_views.xml [hunk 8]
```diff
@@ -495,22 +512,22 @@
             <field name="name">event.registration.tree</field>
             <field name="model">event.registration</field>
             <field name="arch" type="xml">
-                <tree string="Registration" multi_edit="1" sample="1">
+                <tree string="Registration" multi_edit="1" sample="1" expand="1" default_order="create_date desc" class="o_event_registration_view_tree">
```
Comment (tde-banana-odoo) [inline addons/event/views/event_views.xml:515]: Linting: Would multi-line definition to ease reading :)

File: addons/event/views/event_views.xml [hunk 9]
```diff
@@ -495,22 +512,22 @@
...
                     <field name="name"/>
                     <field name="partner_id" optional="hide"/>
                     <field name="email" optional="show"/>
                     <field name="phone" optional="show"/>
                     <field name="mobile" optional="hide"/>
                     <field name="event_id" invisible="context.get('default_event_id')"/>
                     <field name="event_ticket_id" domain="[('event_id', '=', event_id)]"/>
-                    <field name="state" readonly="0"/>
+                    <field name="activity_ids" widget="list_activity"/>
+                    <field name="state" readonly="0" decoration-info="state in ('draft', 'open')" decoration-success="state == 'done'" decoration-muted="state == 'cancel'" widget="badge"/>
```
Comment (tde-banana-odoo) [inline addons/event/views/event_views.xml:525]: Linting: Would multi-line definition to ease reading :)

File: addons/event/views/event_views.xml [hunk 10]
```diff
@@ -364,7 +381,7 @@
                                             <div t-if="record.address_id.value"><i class="fa fa-map-marker" title="Location"/> <span class="o_text_overflow o_event_kanban_location" t-esc="record.address_id.value"/></div>
                                         </div>
                                         <h5 class="o_event_fontsize_11 p-0">
-                                            <a name="%(act_event_registration_from_event)d" type="action" context="{'search_default_expected': True}">
+                                            <a name="%(act_event_registration_from_event)d" type="action" context="{'search_default_expected': True, 'search_default_group_by_create_date_week': True}">
```
Comment (tde-banana-odoo) [inline addons/event/views/event_views.xml:384]: Linting: Would multi-line definition to ease reading :)

File: addons/event/views/event_views.xml [hunk 11]
```diff
@@ -181,7 +198,7 @@
                         <div class="oe_button_box" name="button_box" groups="base.group_user">
                             <button name="%(event.act_event_registration_from_event)d"
                                     type="action"
-                                    context="{'search_default_expected': True}"
+                                    context="{'search_default_expected': True, 'search_default_group_by_create_date_week': True}"
```
Comment (tde-banana-odoo) [inline addons/event/views/event_views.xml:201]: Linting: Would multi-line definition to ease reading :)

File: addons/event/views/event_views.xml (hunk 9 above)
Comment (tde-banana-odoo) [inline addons/event/views/event_views.xml:525]: Do we need the readonly="0" since it is not readonly at all ?

File: addons/event/models/event_registration.py [hunk 12]
```diff
@@ -27,17 +27,14 @@ class EventRegistration(models.Model):
     utm_source_id = fields.Many2one('utm.source', 'Source', index=True, ondelete='set null')
     utm_medium_id = fields.Many2one('utm.medium', 'Medium', index=True, ondelete='set null')
     # attendee
-    partner_id = fields.Many2one(
-        'res.partner', string='Booked by',
-        states={'done': [('readonly', True)]})
+    partner_id = fields.Many2one('res.partner', string='Booked by', tracking=True)
```
Comment (tde-banana-odoo) [inline addons/event/models/event_registration.py:30]: Could probably set tracking to 1 ? That way we correctly order them :)

File: addons/website_event_questions/models/event_registration.py [hunk 13]
```diff
@@ -28,3 +28,10 @@ class EventRegistrationAnswer(models.Model):
     _sql_constraints = [
         ('value_check', "CHECK(value_answer_id IS NOT NULL OR COALESCE(value_text_box, '') <> '')", "There must be a suggested value or a text value.")
     ]
+
+    # for displaying selected answers by attendees in attendees list view
+    def name_get(self):
+        res = []
+        for rec in self:
+            res.append((rec.id, ("%s" % rec.value_answer_id.name if rec.question_type == "simple_choice" else rec.value_text_box)))
```
Comment (tde-banana-odoo) [inline addons/website_event_questions/models/event_registration.py:36]: Linting: would make a list comprehension + use ``reg`` (as registration) + multi-line to ease reading,like
```
return [
    (reg.id,
     reg.value_answer_id.name if reg.question_type == "simple_choice" else reg.value_text_box
    )
    for reg in self
]
```

File: addons/event/views/event_views.xml [hunk 14]
```diff
@@ -714,8 +732,15 @@
...
             <field name="domain"></field>
-            <field name="view_mode">pivot,graph,kanban,tree,form</field>
-            <field name="context">{}</field>
+            <field name="view_mode">graph,pivot,kanban,tree,form</field>
+            <field name="context">{
+                    'search_default_filter_last_month_creation': 1,
+                    'search_default_status': 1,
+                    'search_default_group_by_create_date_day': 1,
+                    'search_default_group_event': 1,
+                    'attendees_from_reporting': 1,
```
Comment (tde-banana-odoo) [inline addons/event/views/event_views.xml:741]: Mmmh ... ``registration_view_hide_group_by_create_date_week`` maybe ? A bit long ... but more explicit. Don't like the "from reporting" as it may be used outside of reporting (action used in other context).

File: addons/event/views/event_views.xml [hunk 15]
```diff
@@ -495,22 +512,22 @@
...
                     <field name="event_ticket_id" domain="[('event_id', '=', event_id)]"/>
-                    <field name="state" readonly="0"/>
+                    <field name="activity_ids" widget="list_activity"/>
+                    <field name="state" readonly="0" decoration-info="state in ('draft', 'open')" decoration-success="state == 'done'" decoration-muted="state == 'cancel'" widget="badge"/>
                     <field name="company_id" groups="base.group_multi_company" optional="hide"/>
                     <field name="message_needaction" invisible="1"/>
                     <button name="action_confirm" string="Confirm" states="draft" type="object" icon="fa-check"/>
                     <button name="action_set_done" string="Mark as Attending" states="open" type="object" icon="fa-level-down"/>
-                    <button name="action_cancel" string="Cancel" states="draft,open" type="object" icon="fa-times"/>
+                    <button name="action_cancel" string="Cancel" states="draft,open" type="object" class="o_btn_cancel_registration" icon="fa-times"/>
```
Comment (tde-banana-odoo) [inline addons/event/views/event_views.xml:530]: Linting: Would multi-line definition to ease reading :)

**Review by tde-banana-odoo — COMMENTED** [review]
Review! Thanks for the PR, looks good :) . Some comments, mainly linting / code polishing.

Cheers !

File: addons/event/tests/common.py [hunk 16]
```diff
@@ -100,9 +100,11 @@ def setUpClass(cls):
         cls.event_0.company_id.write({'country_id': cls.env.ref('base.be').id})
 
     @classmethod
-    def _create_registrations(cls, event, reg_count):
+    def _create_registrations(cls, event, reg_count, create_date=False):
```
Comment (awa-odoo) [inline addons/event/tests/common.py:103]: Do you actually use this new parameter anywhere?

File: addons/event/tests/test_event_mail_schedule.py [hunk 17]
```diff
@@ -89,12 +89,14 @@ def test_event_mail_schedule(self):
 
         # create some registrations
         with freeze_time(now), self.mock_mail_gateway():
-            reg1 = self.env['event.registration'].with_user(self.user_eventuser).create({
+            reg1 = self.env['event.registration'].create({
+                'create_date': now,
```
Comment (awa-odoo) [inline addons/event/tests/test_event_mail_schedule.py:93]: Don't think you have to specify the create_date, the ORM should have its time frozen and compute it correctly.
Same applies for other create calls.

File: addons/event/views/event_views.xml [hunk 18]
```diff
@@ -554,7 +553,8 @@
                                     ]"
                                     attrs="{'invisible': [('event_id', '=', False)]}"
                                     options="{'no_open': True, 'no_create': True}"/>
-                                <field name="date_open" groups="base.group_no_one"/>
+                                <field name="partner_id"/>
```
Comment (awa-odoo) [inline addons/event/views/event_views.xml:577]: Why do we add partner_id here? It's already in the "Attendee" tab.
Also seems a bit unrelated to this commit.

File: addons/event/views/event_views.xml [hunk 19]
```diff
@@ -692,7 +692,6 @@
                     <filter string="Confirmed" name="confirmed" domain="[('state', '=', 'open')]"/>
                     <filter string="Attended" name="attended" domain="[('state', '=', 'done')]"/>
                     <separator/>
-                    <filter string="Registration Date" name="filter_date_open" date="date_open"/>
```
Comment (awa-odoo) [inline addons/event/views/event_views.xml:689]: Shouldn't we replace it with a filter on "create_date" ?

File: addons/event/views/event_views.xml (hunk 4 above)
Comment (awa-odoo) [reply addons/event/views/event_views.xml:536]: I think there is indeed a mismatch in the commits, the field "partner_id" is added (and duplicated) in an earlier commit, then removed in this commit.

File: addons/event/views/event_views.xml [hunk 20]
```diff
@@ -135,6 +135,23 @@
             </field>
         </record>
 
+        <!-- We need kanban view to be displayed first while coming from registration desk, so we have created
+         new action and changed the view sequence. -->
+        <record id="event_registration_action_kanban" model="ir.actions.act_window">
```
Comment (awa-odoo) [inline addons/event/views/event_views.xml:140]: I see that you created this new action but... where is it used?

**Review by awa-odoo — COMMENTED** [review]
Hello @ija-odoo 

Nice little views rework.
I added a few extra comments, can you take a look?

Thanks & cheers!

File: addons/event/tests/test_event_mail_schedule.py (hunk 17 above)
Comment (tde-banana-odoo) [reply addons/event/tests/test_event_mail_schedule.py:93]: Well you end up with create date linked to SQL date. I had to do the same trick elsewhere ...

File: addons/event/views/event_views.xml (hunk 20 above)
Comment (tde-banana-odoo) [reply addons/event/views/event_views.xml:140]: Enterprise ... I think it is better to keep it here in case we want to use it someday (and avoid yet another action creation).

File: addons/event/views/event_views.xml (hunk 19 above)
Comment (ija-odoo) [reply addons/event/views/event_views.xml:689]: yes, we can replace it with create_date.

File: addons/event/views/event_views.xml (hunk 18 above)
Comment (ija-odoo) [reply addons/event/views/event_views.xml:577]: Accidently added to a wrong commit.

File: addons/event/tests/common.py (hunk 16 above)
Comment (ija-odoo) [reply addons/event/tests/common.py:103]: No, we don't use this parameter, i will remove it.

**Comment by tde-banana-odoo** [comment]
@robodoo r+ rebase-merge
